bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryDirectDataWriterHelper.java

```diff
@@ -63,6 +69,10 @@ public class BigQueryDirectDataWriterHelper {
   private long appendRequestSizeBytes = 0; // number of bytes waiting for the next append request
   private long writeStreamRowCount = 0; // total offset / rows of the current write-stream
 
+  private final ExecutorService appendRowsExecutor = Executors.newSingleThreadExecutor();
+  private final List<ApiFuture<AppendRowsResponse>> appendRowsFutures = new ArrayList<>();
```

Comment:
This use case seems perfect for a queue. FIFO usage. We can get rid of nextIndexCheckResponse.

Reply:
Done.